apps/namadillo/src/App/Ibc/IbcTransfer.tsx

```diff
@@ -1,172 +1,227 @@
+        channelId: sourceChannelId,
+      },
+    });
+  };
 
   return (
-    <Panel title="IBC" className="mb-2 bg-[#999999] text-black">
-      {/* Error */}
-      <p className="text-[#ff0000]">{error}</p>
-
-      <hr />
-
-      {/* Keplr addresses */}
-      <h3>Keplr addresses</h3>
-      <button className={buttonStyles} onClick={getAddress}>
-        get address
-      </button>
-      <p>
-        {alias} {address}
-      </p>
-
-      <hr />
-
-      {/* Balances */}
-      <h3>Balances</h3>
-      <button className={buttonStyles} onClick={getBalances}>
-        get balances
-      </button>
-      {balances?.map(({ denom, amount }) => (
-        <div key={denom}>
-          <label>
-            <input
-              type="radio"
-              name="token"
-              value={denom}
-              checked={token === denom}
-              onChange={(e) => setToken(e.target.value)}
-            />
-            {denom} {amount}
-          </label>
-        </div>
-      ))}
-
-      <hr />
-
-      {/* Namada accounts */}
-      <h3>Namada accounts</h3>
-      <button className={buttonStyles} onClick={getNamadaAccounts}>
-        get namada accounts
-      </button>
-
-      {namadaAccounts?.map(({ alias, address }) => (
-        <div key={address}>
-          <label>
-            <input
-              type="radio"
-              name="target"
-              value={address}
-              checked={target === address}
-              onChange={(e) => setTarget(e.target.value)}
-            />
-            {alias} {address}
-          </label>
-        </div>
-      ))}
-
-      <hr />
-
-      {/* Amount to send */}
-      <h3>Amount to send</h3>
-      <input value={amount} onChange={(e) => setAmount(e.target.value)} />
-
-      <hr />
-
-      {/* Channel ID */}
-      <h3>Channel ID</h3>
-      <input value={channelId} onChange={(e) => setChannelId(e.target.value)} />
-
-      <hr />
-
-      {/* Submit IBC transfer */}
-      <h3>Submit IBC transfer</h3>
-      <button className={buttonStyles} onClick={submitIbcTransfer}>
-        submit IBC transfer
-      </button>
+    <Panel>
+      <header className="text-center mb-4">
+        <h2>IBC Transfer to Namada</h2>
+      </header>
+
+      <input
+        className="text-black"
+        type="text"
+        placeholder="source channel id"
+        value={sourceChannelId}
+        onChange={(e) => setSourceChannelId(e.target.value)}
+      />
+
+      <TransferModule
+        source={{
+          isLoadingAssets: assetBalanceQuery.isFetching,
+          availableAssets:
+            Object.values(assetsBalances).map((el) => el.asset) || [],
+          selectedAsset,
+          onChangeSelectedAsset: setSelectedAsset,
+          availableAmount:
+            selectedAsset && selectedAsset.base in assetsBalances ?
+              assetsBalances[selectedAsset.base].balance
+            : undefined,
+          availableChains: Object.values(knownChains).map(
```

Comment:
I think we will need to deal with the situation where one or more of the chains we want to list is not known by the user's Keplr wallet. For example, at the moment I am getting the error `There is no chain info for mocha-4` because I am missing that chain in Keplr.

We can get the chains in Keplr using `keplr.getChainInfosWithEndpoints`. I'm not sure how we should handle this situation though. We could just not display unknown chains, or disable them, or even prompt them to be added to Keplr.

Reply:
That's a good point. I wonder if we should even show any chains not currently enabled by the user's wallet, since we wouldn't be able to get those addresses or sign unless we prompt them to first enable that chain. Hmmm

Reply:
Just made a few changes giving the possibility of adding an unknown chain to the Keplr wallet. This will be useful mainly for the testnet chains.